ui/packages/catalog-ui-search/src/main/webapp/js/widgets/cesium.polygon.js

```diff
@@ -9,127 +9,185 @@
-                }
-                this.remove(); // backbone cleanup.
+const Marionette = require('marionette');
+const Cesium = require('cesium');
+const _ = require('underscore');
+const wreqr = require('wreqr');
+const NotificationView = require('./notification.view');
+const ShapeUtils = require('js/ShapeUtils');
+const DrawingController = require('./drawing.controller');
+const Turf = require('@turf/turf');
+const DistanceUtils = require('js/DistanceUtils');
+
+"use strict";
+
+const THRESHOLD = 8000000;
+
+const createBufferedPolygonPoints = (polygonPoints, model) => {
+    const width = DistanceUtils.getDistanceInMeters(model.toJSON().polygonBufferWidth, model.get('polygonBufferUnits')) || 1;
+
+    return Turf.buffer(Turf.lineString(polygonPoints), Math.max(width, 1), 'meters');
+}
+
+const getCurrentMagnitudeFromView = (view) => {
+    return view.options.map.camera.getMagnitude();
+}
+
+const needsRedraw = (view) => {
+    const currentMagnitude = getCurrentMagnitudeFromView(view);
+
+    if (view.cameraMagnitude < THRESHOLD && currentMagnitude > THRESHOLD) {
+        return true;
+    }
+    if (view.cameraMagnitude > THRESHOLD && currentMagnitude < THRESHOLD) {
+        return true;
+    }
+
+    return false
+}
+
+const Draw = {};
+
+Draw.PolygonRenderView = Marionette.View.extend({
+    cameraMagnitude: undefined,
+    animationFrameId: undefined,
+    initialize: function() {
+        this.listenTo(this.model, 'change:polygon change:polygonBufferWidth, change:polygonBufferUnits', this.updatePrimitive);
+        this.updatePrimitive(this.model);
+        this.listenForCameraChange();
+    },
+    modelEvents: {
+        'changed': 'updatePrimitive'
+    },
+    updatePrimitive() {
+        this.drawPolygon(this.model);
+    },
+    drawPolygon(model) {
+        const json = model.toJSON();
+        const isMultiPolygon =  ShapeUtils.isArray3D(json.polygon);
+        const polygons = isMultiPolygon ? json.polygon : [ json.polygon ];
+
+        const color = this.model.get('color');
+
+        // first destroy old one
+        if (this.primitive && !this.primitive.isDestroyed()) {
+            this.options.map.scene.primitives.remove(this.primitive);
+        }
+
+        this.primitive = new Cesium.PolylineCollection();
+        this.cameraMagnitude = this.options.map.camera.getMagnitude();
+
+        (polygons || []).forEach(function(polygonPoints){
+            if (!polygonPoints || polygonPoints.length < 3) {
+                return;
+            }
+            if (polygonPoints[0].toString() !== polygonPoints[polygonPoints.length - 1].toString()) {
+                polygonPoints.push(polygonPoints[0]);
+            }
+            const bufferedPolygonPoints = createBufferedPolygonPoints(polygonPoints, this.model);
+            const primitive = this.primitive;
+            bufferedPolygonPoints.geometry.coordinates.forEach(set => 
+                primitive.add({
+                    width: 8,
```

Comment:
❓ Is there a way to pull values like `width` and `outlineWidth` out to make them configurable?

Reply:
Yes we can, though we will need to add the settings to either admin ui or user preferences settings first and also address this for all other drawing components.